smartsim/_core/_install/builder.py

```diff
@@ -385,10 +395,21 @@ def __init__(
         self.libtf_dir = libtf_dir
         self.torch_dir = torch_dir
 
-        # TODO: It might be worth making these constructor args so that users
-        #       of this class can configure exactly _what_ they are building.
-        self._os = OperatingSystem.from_str(platform.system())
-        self._architecture = Architecture.from_str(platform.machine())
+        # Sanity checks
+        self._check_backends_arm64()
+
+    def _check_backends_arm64(self) -> None:
+        if self._architecture == Architecture.ARM64:
+            unsupported = []
+            if self.build_tf:
+                unsupported.append("Tensorflow")
+            if self.build_onnx:
+                unsupported.append("ONNX")
```

Comment:
Should these checks be on `self.build_{tf,onnx}` or `self.fetch_{tf,onnx}`. Is there a reason we would want to try and stop someone from compiling and trying to use their own ARM64 version of TF (or onnx)?

Reply:
Oh yes good call